### PR TITLE
GPU Type

### PR DESCRIPTION
**Pull request by wild-endeavor**
# TL;DR
Adds support for passing along extra metadata related to resources (e.g. accelerator type) to the backend

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

**Review by hamersaw — APPROVED** [review]

**Review by pingsutw — APPROVED** [review]

File: protos/flyteidl/core/tasks.proto [hunk 1]
```diff
@@ -61,6 +61,22 @@ message RuntimeMetadata {
...
+    string device = 1;
+    oneof partition_size_value {
+        bool unpartitioned = 2;
+        string partition_size = 3;
+    }
+}
+
+// Additional metadata associated with resources to allocate to a task
+message ResourceMetadata {
+    oneof accelerator_value {
```
Comment (EngHabu) [inline protos/flyteidl/core/tasks.proto:75]: Why oneof? if it's not set, it'll just be nil, do you want to allow nullifying a field? if so maybe that should go into the override message?

Comment (jeevb) [reply protos/flyteidl/core/tasks.proto:75]: The thought was to support other accelerator types like TPUs in the future. One option would have been to just add all possible accelerator types as separate fields without the guarantee that oneof provides - that only one type is set at any given point in time.

I actually did not consider nullifying an accelerator specified in the task metadata, in the override. That does sound nice. If we were to do that, would it be a bad idea to add a null_accelerator to the oneof?

Comment (EngHabu) [reply protos/flyteidl/core/tasks.proto:75]: Since oneofs do not change wire format, I would keep it out and kick the can down the road when we add a different accelerator to decide then.

There isn't a great way to nullify in Protos unfortunately...

Comment (jeevb) [reply protos/flyteidl/core/tasks.proto:75]: https://github.com/flyteorg/flyteidl/pull/442
